alphabase/io/tempmmap.py

```diff
@@ -165,7 +165,7 @@ def array(shape: tuple, dtype: np.dtype, tmp_dir_abs_path: str = None) -> np.nda
 
     with h5py.File(temp_file_name, "w") as hdf_file:
         array = hdf_file.create_dataset("array", shape=shape, dtype=dtype)
-        array[0] = 0
+        array[0] = np.string_("") if isinstance(dtype, np.dtypes.ObjectDType) else 0
```

Comment:
looking at this again .. could you also use `StrDType` here? https://numpy.org/devdocs/reference/routines.dtypes.html#numpy.dtypes.StrDType

Reply:
done